File: manager/src/main/java/moe/shizuku/manager/utils/ClipboardUtils.java
package moe.shizuku.manager.utils;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;

public final class ClipboardUtils {

    private ClipboardUtils() {
    }

    public static boolean put(Context context, CharSequence text) {
        return put(context, ClipData.newPlainText("label", text));
    }

    public static boolean put(Context context, ClipData clipData) {
        try {
            ((ClipboardManager) context.getSystemService("clipboard")).setPrimaryClip(clipData);
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: manager/src/main/java/eu/darken/porter/manager/utils/ClipboardUtils.kt
package eu.darken.porter.manager.utils

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context

object ClipboardUtils {

    fun put(context: Context, text: CharSequence): Boolean = put(context, ClipData.newPlainText("label", text))

    fun put(context: Context, clipData: ClipData): Boolean = try {
        (context.getSystemService("clipboard") as ClipboardManager).setPrimaryClip(clipData)
        true
    } catch (e: Exception) {
        false
    }
}
